### PR TITLE
pypi_formula_mappings: extra_packages for python@X

### PR DESCRIPTION
**Pull request by woodruffw**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

I'm not 100% confident that this is the "right" way to do things, but this _should_ address cases like https://github.com/Homebrew/homebrew-core/pull/153070 and https://github.com/Homebrew/homebrew-core/pull/153069 -- when `python@X` is bumped these resources appear to be required, so we mark them as such in the PyPI formula mapping.

### CODE REVIEW

**Comment by woodruffw** [comment]
CC @chenrui333 for opinions 🙂

File: pypi_formula_mappings.json [hunk 1]
```diff
@@ -844,6 +844,24 @@
...
+  },
+  "python@3.10": {
+    "extra_packages": ["setuptools", "pip", "wheel"]
+  },
+  "python@3.11": {
+    "extra_packages": ["setuptools", "pip", "wheel"]
+  },
+  "python@3.12": {
+    "extra_packages": ["setuptools", "pip", "wheel"]
+  },
```
Comment (chenrui333) [inline pypi_formula_mappings.json:864]: you also need `flit-core`

Comment (p-linnane) [reply pypi_formula_mappings.json:864]: We can also leave out 3.7 since it's end of life and disabled.

Comment (chenrui333) [inline pypi_formula_mappings.json:864]:
```suggestion
  "python@3.8": {
    "extra_packages": ["flit-core", "pip", "setuptools", "wheel"]
  },
  "python@3.9": {
    "extra_packages": ["flit-core", "pip", "setuptools", "wheel"]
  },
  "python@3.10": {
    "extra_packages": ["flit-core", "pip", "setuptools", "wheel"]
  },
  "python@3.11": {
    "extra_packages": ["flit-core", "pip", "setuptools", "wheel"]
  },
  "python@3.12": {
    "extra_packages": ["flit-core", "pip", "setuptools", "wheel"]
  },
```

Comment (chenrui333) [reply pypi_formula_mappings.json:864]: 👍

Comment (woodruffw) [reply pypi_formula_mappings.json:864]: > you also need `flit-core`

I saw that in some of the versions but not others; do you know where it came from? AFAICT Python itself doesn't use `flit-core` anywhere in its build.

Comment (chenrui333) [reply pypi_formula_mappings.json:864]: @woodruffw there is come context in this thread:

https://github.com/Homebrew/homebrew-core/pull/127641
https://github.com/pypa/wheel/pull/501

**Comment by woodruffw** [comment]
Added `flit-core`, dropped 3.7 from mapping.

**Review by chenrui333 — APPROVED** [review]

**Comment by p-linnane** [comment]
@woodruffw 3.8 does not need `flit-core`, but good to merge this otherwise.

**Comment by woodruffw** [comment]
> @woodruffw 3.8 does not need `flit-core`, but good to merge this otherwise.

Dropped, thank you!

**Review by p-linnane — APPROVED** [review]
@woodruffw Thanks!

**Pull request merged**